src/ssh/azext_ssh/ssh_info.py

```diff
@@ -204,3 +249,45 @@ def _create_relay_info_file(self):
             logger.warning("Couldn't determine relay information expiration. Error: %s", str(e))
 
         return relay_info_path
+    
+     # Passing in the resource Tags to the SSHSession object
+    def set_azure_resource_tags(self, tags):
+        self.azure_resource_tags = tags
+        return
+
+    # Configuring the port from the resource tag and validating the port number
+    def configure_port_from_resource_tag(self):
+        logger = log.get_logger(__name__)
+
+        if self.port and self.resource_tag:
+            logger.warning("Warning: Both --port and --resource-tag arguments were specified."
+                            "The --port option will take precedence and the --resource-tag will be ignored."
+                            "To use the port number from the --resource-tag, please omit the --port argument.")
+            return
+        if self.port:
+            return
+        
+        # Validating the port number and setting the port to the port value from the resource tag
+        def validate_and_set_port(tag_name="SSHPort"):
+            port_num = self.azure_resource_tags.get(tag_name)
+            if port_num and port_num.isdigit() and int(port_num) < 65535:
+                self.port = port_num
+
+            else:
+                raise azclierror.InvalidArgumentValueError(
+                    f"Port '{port_num}' from resource tag '{tag_name}' is not supported by this command."
+                    "Port numbers must not be empty, must not contain letters or special characters, and cannot exceed 65535 port value."
+                    "Please contact your administrator to correct the resource tag value or use the --port parameter. ")
+            
+
+        if self.resource_tag:
+            if self.resource_tag in self.azure_resource_tags:
+                validate_and_set_port(self.resource_tag)
+            else:
+                raise azclierror.InvalidArgumentValueError(
+                f"Resource tag name '{self.resource_tag}' cannot be found. Contact your administrator to ensure the tag is valid."
+            )
+
+        elif "SSHPort" in self.azure_resource_tags:
+            validate_and_set_port()
+        return
```

Comment:
The fact that we need to repeat all this code for both classes is not great. Can you look into just having a parent class with the common properties and methods, and having ConfigSession and SSHSession inherit from that class? I think we never did it in the past because the methods in each class were mostly different, so I didn't bother at the time, but now there is a lot of repeated code. You could create an abstract parent class called something like BaseSession and have SSHSession and ConfigSession inherit from that class